### DISCUSSION TITLE
Missing SWO documentation/example

### DISCUSSION THREAD

**Discussion by boraozgen** [pyocd/pyOCD#1238] [General]
Hello,

I am trying to read SWO output from an STM32F4 controller and ST-Link probe. I can already receive the SWO output using the J-Link SWO tool and its python interface [pylink](https://github.com/square/pylink).

The only documentation I found about SWO and pyOCD is this PR: https://github.com/pyocd/pyOCD/pull/543

It is suggested to start the GDB server with the correct options. I could not get it running consistently, I believe it has something to do with the SWO clock speed. I think pyOCD requires the SWO clock prescaler to be set correctly by the firmware. I believe it is a better approach to set this register from the tool to avoid clock mismatch.

I would also like to read the SWO output from the Python API. I could not find any information regarding this.

Could someone point to an example to use SWO? If pyOCD is not meant to read the SWO output, are there any suitable alternatives?

**Comment by flit** [comment]
Hi @boraozgen. You are certainly correctly that there is little to no documentation for enabling SWO in pyocd. I do plan on adding documentation as soon as possible (adding better docs is pretty high priority now the pyocd.io site is up).

Short-term, and as a precursor to the docs, I'll set up an STM32 based test so I can verify all the steps and report back here. (It's been a few months since I last used SWO.)
